kernel/src/boot_stage2.rs

```diff
@@ -98,19 +110,18 @@ global_asm!(
         orl %edx, %eax
         movl %eax, 0xF6C(%edi)
 
-        /* Signal APs */
-        movl $setup_flag, %edi
-        movl $1, (%edi)
         jmp 2f
 
-.Lskip_paging_setup:
-        movl $setup_flag, %edi
-.Lap_wait:
+    .Lskip_paging_setup:
+        movl $ap_flag, %edi
+    .Lap_wait_for_env:
         movl (%edi), %eax
         test %eax, %eax
```

Comment:
```suggestion
        cmpl (%edi), $0
```

It's better to perform the comparison in a single instruction instead of load followed by a compare.  This code predates your change, but you may as well clean it up while you're in this code.

Reply:
Sure I'll clean that up.